neuralmonkey/dataset.py

```diff
@@ -417,13 +268,185 @@ def add_series(self, name: str, series: Iterable[Any]) -> None:
             "Lazy dataset does not support adding series.")
 
 
-def _preprocessed_datasets(dataset: Dataset, kwargs: Dict[str, Any]) -> None:
+# pylint: disable=invalid-name
+DatasetPreprocess = Callable[[Dataset], Iterable[Any]]
+DatasetPostprocess = Callable[[Dataset, Dict[str, Iterable[Any]]],
+                              Iterable[Any]]
+ReaderDef = Union[str, List[str],
+                  Tuple[str, Reader], Tuple[List[str], Reader]]
+Kwargs = Dict[str, Union[ReaderDef, DatasetPreprocess]]
```

Comment:
V tom případě bych to pojmenoval nějak rozumně podle toho, co je uložený vevnitř, třeba InputSeriesMapping nebo tak